src/Components/GEOSldas_GridComp/GEOSlandassim_GridComp/clsm_ensupd_read_obs.F90

```diff
@@ -4584,6 +4585,534 @@ subroutine read_obs_SMOS( date_time, N_catd, this_obs_param,      &
+    integer                      :: dim_sizes(2)
+    integer                      :: start(2), edges(2), stride(2)
+    character(100)               :: var_name
+
+    integer                      :: status, n_read, record_pos
+    
+    integer,           parameter :: DFACC_READ     =  1 ! from hdf.inc
+    integer,           parameter :: DFNT_UINT8     = 21
+    integer,           parameter :: FULL_INTERLACE =  0 ! from hdf.inc
+ 
+    logical                      :: keep_data
+
+    ! variables to define latitude and longitude    
+
+    integer                     :: i, j, k, k_off, ll, mm, kk, L
+    integer                     :: time_index
+
+    real,             parameter :: bin_size = 0.05
+    integer,          parameter :: XGRID    = 3600
+    integer,          parameter :: YGRID    = 7200
+
+    integer, dimension(N_files) :: N_data_tmp
+
+    real                        :: lat_ind(XGRID) = (/(ll, ll=0, XGRID-1, 1)/)
+    real                        :: lon_ind(YGRID) = (/(mm, mm=0, YGRID-1, 1)/)
+    real,      dimension(XGRID) :: lat_c(XGRID)
+    real,      dimension(YGRID) :: lon_c(YGRID)
+    real,dimension(XGRID*YGRID) :: lat_1D, lon_1D
+    real                        :: lon_subtime(9) = (/(kk, kk=180, -180, -45)/)
+     
+    integer, dimension(:), allocatable :: Snow_QA, CI_Index, Cloud_Index 
+    character(len=*),        parameter :: Iam = 'read_MODISscf_hdf'
+    character(len=400)                 :: err_msg
+
+    ! -------------------------------------------------------------------------
+
+    ! initialize N_data
+
+    N_data_tmp(N_files) = XGRID*YGRID      !?!?!?! WHY ONLY FOR LAST ELEMENT????
+    N_data              = XGRID*YGRID 
+
+    ! Define latitude and longitude 
+    !
+    ! VERIFY THAT lat/lon DEFINITION HERE IS CONSISTENT WITH lat/lon STORED IN FILE ?????
+    ! PER FILE SPECS: 
+    !   Lat = Upper left X coordinate for each grid cell in degrees north
+    !   Lon = Upper left Y coordinate for each grid cell in degrees east
+
+    lat_c   = (  90-bin_size/2)-bin_size*lat_ind       
+    lon_c   = (-180+bin_size/2)+bin_size*lon_ind
+
+    ! Create a 1D structure for lat and lon (could change to remove this)
+
+    do i=1,7200
+
+      lat_1D(3600*(i-1)+1:3600*i)= lat_c        
+      lon_1D(3600*(i-1)+1:3600*i)= lon_c(i)
+
+    end do 
+
+    ! Ensure output pointers are not allocated (must be deallocated outside this subroutine)
+
+    if ( associated(MODIS_lon) .or. associated(MODIS_lat) .or. associated(MODIS_SCF) ) then 
+
+       err_msg = 'output pointers must not be associated/allocated on input.'
+       call ldas_abort(LDAS_GENERIC_ERROR, Iam, err_msg)
+       
+    end if
+    
+    ! Allocate SCF and QC vectors
+
+    allocate(SCF_raw(    N_data))    
+    allocate(CI_Index(   N_data))
+    allocate(Snow_QA(    N_data))
+    allocate(Cloud_Index(N_data))
+
+    ! read hdf data into arrays, concatenate data from N_files files   !?!?!?!?! NOT HAPPENING...
```

Comment:
Not sure if we need the functionality to read more than one file at a given analysis time.  If so, the current reader does not provide this functionality.  If it is not needed, then the reader needs to be cleaned up so it's obvious that only one file is read.